src/main/kotlin/com/ampnet/reportserviceeth/service/impl/IpfsServiceImpl.kt

```diff
@@ -0,0 +1,45 @@
+package com.ampnet.reportserviceeth.service.impl
+
+import com.ampnet.reportserviceeth.config.ApplicationProperties
+import com.ampnet.reportserviceeth.service.IpfsService
+import mu.KLogging
+import org.springframework.stereotype.Service
+import org.springframework.web.client.RestClientException
+import org.springframework.web.client.RestTemplate
+import org.springframework.web.client.postForEntity
+import org.springframework.web.util.UriComponentsBuilder
+
+@Service
+class IpfsServiceImpl(
+    private val applicationProperties: ApplicationProperties,
+    private val restTemplate: RestTemplate
+) : IpfsService {
+
+    companion object : KLogging()
+
+    override fun getLogoHash(issuerInfoHash: String): String? {
+        val request = UriComponentsBuilder.fromUriString(applicationProperties.ipfs.frontendApi)
+            .queryParam("hash", issuerInfoHash).build().toUri()
+        try {
+            val response = restTemplate.postForEntity<IssuerInfoResponse>(request)
+            return if (response.statusCode.is2xxSuccessful) {
```

Comment:
Add a hashmap to store successfully fetched logos. Return the logo if it is stored in the map.